packages/metal-dom/src/dom.js

```diff
@@ -306,28 +306,32 @@ class dom {
 	 * @protected
 	 */
 	static handleDelegateEvent_(event) {
-		dom.normalizeDelegateEvent_(event);
 		var currElement = core.isDef(event[NEXT_TARGET]) ?
-			event[NEXT_TARGET] :
-			event.target;
+				event[NEXT_TARGET] :
+				event.target;
 		var ret = true;
-		var container = event.currentTarget;
-		var limit = event.currentTarget.parentNode;
-		var defFns = [];
 
-		while (currElement && currElement !== limit && !event.stopped) {
-			event.delegateTarget = currElement;
-			ret &= dom.triggerMatchedListeners_(container, currElement, event, defFns);
-			currElement = currElement.parentNode;
-		}
+		if (dom.isAbleToInteractWith_(currElement, event.type)) {
```

Comment:
This diff seems kind of messy but I've only written this `if` statement.

In my opinion, this method is becoming difficult to read. Maybe we can make improvements on it. I have some suggestions:

1 - We can create a new method for checking the isAbleToInteractWith_ condition before this handleDelegateEvent_ method, or, put the logic of the handleDelegateEvent_ in a new one. 

2 - We can keep the checking of isAbleToInteractWith_ here, in this method, but above the code, and then, we can return the default value of the `ret` variable. For me, it is strange to return the same variable in two cases, at the end of the code and at the beginning, but, it is an option.

I would like to hear your opinion @mairatma.